### PR TITLE
Rename js udf to match conventions

### PR DESCRIPTION
**Pull request by jklukas**
Fixup to #200 where I missed @relud's comments before merging.

### CODE REVIEW

File: udf/udf_json_extract_histogram_js.sql [hunk 1]
```diff
@@ -9,7 +9,7 @@ be more usable in practice.
 */
```
Comment (relud) [inline udf/udf_json_extract_histogram_js.sql]: requesting the js be used as an extension: `udf/udf_json_extract_histogram.js.sql`

Comment (jklukas) [reply udf/udf_json_extract_histogram_js.sql]: I should actually read the README carefully before making fixes :)

Comment (jklukas) [reply udf/udf_json_extract_histogram_js.sql]: To support this naming (file is `udf/udf_json_extract_histogram.js.sql` and the name of the udf itself if `udf_json_extract_histogram_js`, we need to alter the test code to replace `.` with `_` when parsing filenames, and we'll need to update the docs to specify that JS udfs should be in files ending in `.js.sql` but with names ending in `_js`.

The above seems more complicated compared to having the filename match the function name `udf_json_extract_histogram_js.sql`, so my preference is to use that convention and update the docs accordingly as part of this PR.

Alternately, we could split JS UDFs to their own `jsudf/` directory and expect that we have functions named like `jsudf_foo` defined in files named `jsudf_foo.sql`. I think this provides a nice bridge for when persistent UDFs go beta; we'd then create `udf` and `jsudf` datasets and deploy functions that are accessed like `udf.json_extract_histogram(foo)` and `jsudf.json_extract_histogram(foo)`.

What do you think?

Comment (relud) [reply udf/udf_json_extract_histogram_js.sql]: > The above seems more complicated compared to having the filename match the function name `udf_json_extract_histogram_js.sql`, so my preference is to use that convention and update the docs accordingly as part of this PR

this seems like a good immediate approach

> Alternately, we could split JS UDFs to their own `jsudf/` directory and expect that we have functions named like `jsudf_foo` defined in files named `jsudf_foo.sql`. I think this provides a nice bridge for when persistent UDFs go beta; we'd then create `udf` and `jsudf` datasets and deploy functions that are accessed like `udf.json_extract_histogram(foo)` and `jsudf.json_extract_histogram(foo)`.

this seems like an excellent long-term solution

File: README.md [hunk 2]
```diff
@@ -42,7 +42,9 @@ Recommended practices
 - Should be used to avoid code duplication
 - Should use lower snake case names e.g. `udf_mode_last`
   - Should use a `udf_` prefix for naming temporary udfs
-- Should be defined in files named as `udfs/function.{sql,js}` e.g. `udfs/udf_mode_last.sql`
+  - Should additionally use a `_js` suffix for naming temporary udfs implemented in JavaScript
+- Should be defined in files named as `udfs/function.sql` e.g. `udfs/udf_mode_last.sql`
+  or `udfs/udf_sum_buckets_with_ci_js.sql`
```
Comment (relud) [inline README.md]: 💯

**Pull request merged**